File: lib-view/src/main/java/com/hx/lib_hx/custom/item_view/KeyValueView.java
package com.hx.lib_hx.custom.item_view;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.hx.lib_hx.R;
import com.hx.lib_hx.utils.DisplayUtils;


public class KeyValueView extends LinearLayout {
    private ImageView mIv;
    private TextView mKey;
    private TextView mValue;
    private TextView mColon;
    public KeyValueView(Context context) {
        this(context,null);
    }

    public KeyValueView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs ,0);
    }

    public KeyValueView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        LayoutInflater.from(context).inflate(R.layout.key_value, this, true);
        mIv = findViewById(R.id.key_icon);
        mKey = findViewById(R.id.key);
        mValue = findViewById(R.id.value);
        mColon = findViewById(R.id.colon);
        initAttrs(context,attrs);
    }

    private void initAttrs(Context context, AttributeSet attrs) {
        TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.KeyValueView);
        if (typedArray != null) {
            int keyColor = typedArray.getColor(R.styleable.KeyValueView_key_text_color, 0xFF424242);
            setKeyColor(keyColor);
            String keyString = typedArray.getString(R.styleable.KeyValueView_key_text);
            setKeyText(keyString);

            int valueColor = typedArray.getColor(R.styleable.KeyValueView_value_text_color, 0xFF000000);
            setValueColor(valueColor);
            String valueString = typedArray.getString(R.styleable.KeyValueView_value_text);
            setValueText(valueString);
            //设置icon
            int resourceId = typedArray.getResourceId(R.styleable.KeyValueView_icon, 0);
            setIconRes(resourceId);
            int iconSize = typedArray.getLayoutDimension(R.styleable.KeyValueView_icon_size, 56);
            setIconSize(iconSize);
            //设置权重
            boolean isAlign_parent = typedArray.getBoolean(R.styleable.KeyValueView_touch_parent_right, false);
            if (isAlign_parent) {
                LayoutParams lp = new LayoutParams(0, LayoutParams.WRAP_CONTENT, 1.0f);
                mValue.setLayoutParams(lp);
                mValue.setGravity(Gravity.RIGHT);
            }
            // 设置冒号
            boolean colonVisi = typedArray.getBoolean(R.styleable.KeyValueView_colon, true);
            setColonVisible(colonVisi);
            float textSize = typedArray.getDimension(R.styleable.KeyValueView_text_size, 12);
            setTextSize(DisplayUtils.sp2px(context,textSize));
            typedArray.recycle();
        }
    }

    private void setColonVisible(boolean colonVisi) {
        mColon.setVisibility(colonVisi?VISIBLE:GONE);
    }

    private void setIconHeight(int iconHeight) {
        ViewGroup.LayoutParams layoutParams = mIv.getLayoutParams();
        layoutParams.height = iconHeight;
        mIv.setLayoutParams(layoutParams);
    }

    private void setIconSize(int iconSize) {
        ViewGroup.LayoutParams layoutParams = mIv.getLayoutParams();
        layoutParams.width = iconSize;
        layoutParams.height = iconSize;
        mIv.setLayoutParams(layoutParams);
    }

    private void setIconRes(int resourceId) {
        mIv.setVisibility(resourceId == 0 ? GONE : VISIBLE);
        mIv.setImageResource(resourceId);
    }


    public void setKeyText(String text) {
        mKey.setText(text);
    }

    public void setValueText(String text) {
        mValue.setText(text);
    }

    public void setTextSize(float size) {
        mKey.setTextSize(size);
        mValue.setTextSize(size);
        mColon.setTextSize(size);
    }

    public void setKeyColor(int color) {
        mKey.setTextColor(color);
    }

    public void setValueColor(int color) {
        mValue.setTextColor(color);
    }
}

File: lib-view/src/main/kotlin/com/kt/lib/custom/KeyValueView.kt
package com.kt.lib.custom

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.hx.lib_hx.R
import com.hx.lib_hx.utils.DisplayUtils

class KeyValueView(context: Context, attrs: AttributeSet?, defStyleAttr: Int)
    : LinearLayout(context, attrs, defStyleAttr) {
    private lateinit var mIv:ImageView
    private lateinit var mKey:TextView
    private lateinit var mValue:TextView
    private lateinit var mColon:TextView

    constructor(context: Context):this(context,null,0)
    constructor(context: Context,attrs: AttributeSet?):this(context,attrs,0)
    constructor(context: Context,defStyleAttr: Int,attrs: AttributeSet?):this(context,attrs,defStyleAttr){
        val view = LayoutInflater.from(context).inflate(R.layout.key_value,this,true)
        mIv = view.findViewById(R.id.key_icon)
        mKey = view.findViewById(R.id.key)
        mValue = view.findViewById(R.id.value)
        mColon = view.findViewById(R.id.colon)
        initAttrs(context,attrs)
    }

    private fun initAttrs(context: Context, attrs: AttributeSet?) {
        val tapeArray = context.obtainStyledAttributes(attrs,R.styleable.KeyValueView)
        if (tapeArray != null) {
            //TextView
            keyColor(tapeArray.getColor(R.styleable.KeyValueView_key_text_color, 0xFF424242.toInt()))
            keyText(tapeArray.getString(R.styleable.KeyValueView_key_text))
            valueColor(tapeArray.getColor(R.styleable.KeyValueView_value_text_color,0xFF000000.toInt()))
            valueText(tapeArray.getString(R.styleable.KeyValueView_value_text))
            //size
            val dimension = tapeArray.getDimension(R.styleable.KeyValueView_text_size, 12.toFloat())
            val size = DisplayUtils.sp2px(context, dimension)
            textSize(size.toFloat())
            //icon
            iconRes(tapeArray.getResourceId(R.styleable.KeyValueView_icon,0))
            //colon
            colonVisible(tapeArray.getBoolean(R.styleable.KeyValueView_colon,true))
        }
        tapeArray.recycle()
    }

    fun colonVisible(visible: Boolean) {
        mColon.visibility = if(visible){ View.VISIBLE}else{View.GONE}
    }

    fun iconSize(size: Int) {
        val layoutParams = mIv.layoutParams
        layoutParams.height=size
        layoutParams.width=size
        mIv.layoutParams=layoutParams
    }

    fun iconRes(res: Int) {
        mIv.setImageResource(res)
    }

    fun textSize(size: Float) {
        mKey.textSize = size
        mValue.textSize = size
        mColon.textSize = size
    }

    fun keyText(text: String) {
        mKey.text=text
    }

    fun valueText(text: String) {
        mValue.text=text
    }

    fun keyColor(color: Int) {
        mKey.setTextColor(color)
    }

    fun valueColor(color: Int) {
        mValue.setTextColor(color)
    }
}